torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -63,18 +59,19 @@ def _put_into_sample(self, sample: Any, id: int, item: Any) -> Any:
 
     def _apply_image_transform(
         self,
-        image: Any,
+        image: Union[torch.Tensor, PIL.Image.Image, features.Image],
         transform_id: str,
         magnitude: float,
         interpolation: InterpolationMode,
-        fill: Union[int, float, Sequence[int], Sequence[float]],
+        fill: Dict[Type, FillType],
     ) -> Any:
 
+        fill_ = fill[type(image)]
         # Fill = 0 is not equivalent to None, https://github.com/pytorch/vision/issues/6517
         # So, we have to put fill as None if fill == 0
         # This is due to BC with stable API which has fill = None by default
-        fill_ = F._geometry._convert_fill_arg(fill)
-        if isinstance(fill, int) and fill == 0:
+        fill_ = F._geometry._convert_fill_arg(fill_)
+        if isinstance(fill_, int) and fill_ == 0:
             fill_ = None
```

Comment:
As discuss on our call, this will revert the BC for when AutoAugment receives `fill=0` and `fill=None` and align the behaviour or prototype and stable.
```suggestion
```

I've raised a separate issue for discussing the potential options for addressing the problem at #6623. Let me know your thoughts.